frontend/src/index.tsx

```diff
@@ -21,20 +21,23 @@ import ReactDOM from "react-dom"
 import { Client as Styletron } from "styletron-engine-atomic"
 import { LightTheme, BaseProvider } from "baseui"
 import { Provider as StyletronProvider } from "styletron-react"
-import { SCSS_VARS } from "autogen/scssVariables"
+import ThemeProvider from "components/core/ThemeProvider"
+import { mainWidgetTheme } from "lib/widgetTheme"
+import { mainTheme } from "theme"
 import AppWithScreencast from "./App"
 
 const engine = new Styletron({ prefix: "st-" })
-const popupZIndex = Number(SCSS_VARS["$z-index-popup-menu"])
 
 ReactDOM.render(
   <StyletronProvider value={engine}>
     {/*
       The BaseProvider type definition doesn't support zIndex, but the object
       actually does. See: https://baseweb.design/components/base-provider/
       // @ts-ignore */}
-    <BaseProvider theme={LightTheme} zIndex={popupZIndex}>
-      <AppWithScreencast />
+    <BaseProvider theme={LightTheme} zIndex={mainTheme.zIndices.popupMenu}>
+      <ThemeProvider theme={mainTheme} baseuiTheme={mainWidgetTheme}>
+        <AppWithScreencast />
+      </ThemeProvider>
```

Comment:
I couldn't see any visual differences so I think it should be fine but just pointing out this applies the baseuiTheme to the main menu and status widget that was not previously applied to. Primarily button + tooltip.

Reply:
Yea. And I think this is good catch to ensure that we conform to some theme. The base theme is set. The only change would be the sidebar theme